server/weaverbird/backends/pipeline_translator/types.py

```diff
@@ -0,0 +1,28 @@
+from typing import List, Callable, Tuple, Protocol, Any, Optional
+
+from pydantic import BaseModel
+from weaverbird.pipeline import Pipeline
+
+
+class StepTranslationReport(BaseModel):
+    step_index: int
+
+
+class PipelineTranslationReport(BaseModel):
+    steps_translation_reports: List[StepTranslationReport]
+
+
+QueryRetriever = Callable[[str], str]
+PipelineTranslator = Callable[[Pipeline, QueryRetriever], Tuple[str, PipelineTranslationReport]]
+
+
+class StepTranslator(Protocol):
+    def __call__(
+        self,
+        step: Any,
+        query: str,
+        query_retriever: Optional[QueryRetriever],
+        translate_pipeline: Optional[PipelineTranslator]
+
+    ):
```

Comment:
What is the return type of a StepTranslator?

Reply:
A string for now. 
It will be updated, when I'll change the data structure forwarded between steps.